cpp/tests/copying/concatenate_tests.cu

```diff
@@ -283,3 +283,116 @@ TEST_F(TableTest, ConcatenateTablesWithOffsetsAndNulls)
     cudf::test::expect_tables_equal(concatenated_tables->view(), table_view_exp1);
   }
 }
+
+struct ListsColumnTest : public cudf::test::BaseFixture {
+};
+
+TEST_F(ListsColumnTest, ConcatenateLists)
+{
+  {
+    cudf::test::lists_column_wrapper a{0, 1, 2, 3};
+    cudf::test::lists_column_wrapper b{4, 5, 6, 7};
+    cudf::test::lists_column_wrapper expected{{0, 1, 2, 3}, {4, 5, 6, 7}};
```

Comment:
In general, it would be good to test with child columns of different lengths. 
Also add in some empty lists. This has tripped up strings concatenate in the past.